Source/JavaScriptCore/wasm/WasmSIMDLaneOperationCounter.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#pragma once
+
+#if ENABLE(WEBASSEMBLY) && ENABLE(B3_JIT)
+
+#include "WasmSIMDOpcodes.h"
+#include <wtf/Atomics.h>
+
+namespace JSC {
+
+class WasmSIMDLaneOperationCounter {
+    WTF_MAKE_FAST_ALLOCATED;
+
+public:
+    static WasmSIMDLaneOperationCounter& singleton();
+
+    void registerAndIncrement(SIMDLaneOperation);
+
+    void dump();
+
+private:
+    constexpr static unsigned m_totalOperationCount = countNumberOfSIMDLaneOperations();
+    Atomic<uint64_t> m_counter[m_totalOperationCount];
```

Comment:
Add a set for Wasm base opcodes, and have `incrementBaseOpcode` increment those.